src/pages/home/report/ContextMenu/ContextMenuActions.tsx

```diff
@@ -542,7 +543,21 @@ const ContextMenuActions: ContextMenuAction[] = [
         icon: Expensicons.Copy,
         successTextTranslateKey: 'reportActionContextMenu.copied',
         successIcon: Expensicons.Checkmark,
-        shouldShow: (type) => type === CONST.CONTEXT_MENU_TYPES.REPORT && (Environment.isDevelopment() || Environment.isStaging() || Environment.isInternalTestBuild()),
+        shouldShow: (type) => {
+            // eslint-disable-next-line react-hooks/rules-of-hooks
+            const {environment, isProduction} = useEnvironment();
+
+            // If we are on production, don't show this option
+            if (isProduction) {
+                return false;
+            }
+
+            const isADHOC = environment === CONST.ENVIRONMENT.ADHOC;
+            const isStaging = environment === CONST.ENVIRONMENT.STAGING;
+            const isDevelopment = environment === CONST.ENVIRONMENT.DEV;
+
+            return type === CONST.CONTEXT_MENU_TYPES.REPORT && (isDevelopment || isStaging || isADHOC);
```

Comment:
I think this code can be simplified:

```suggestion
            return type === CONST.CONTEXT_MENU_TYPES.REPORT && !isProduction);
```

Reply:
Agreed, corrected it